api/services/Domain.js

```diff
@@ -111,6 +111,73 @@ function checkAcmeChallengeDnsRecord(domain) {
   );
 }
 
+/**
+ * @param {SiteModel} site
+ * @param {DomainModel[]} domains
+ * @param {string} context 'site' or 'demo' corresponding to Domain.context
+ */
+function isSiteUrlManagedByDomain(site, domains, context) {
+  const siteDomain = context === 'site' ? 'domain' : 'demoDomain';
+  if (site[siteDomain] === null) { return true; }
+  if (domains.length === 0) { return false; }
+  let siteUrlIsManagedByDomain = false;
+  domains.forEach((domain) => {
+    if (domain.context === context) {
+      if (domain.names.split(',').find(name => `https://${name}` === site[siteDomain])) {
+        siteUrlIsManagedByDomain = true;
+      }
+    }
+  });
+  return siteUrlIsManagedByDomain;
+}
+
+/**
+ * @param {DomainModel} domain
+ */
+async function rebuildAssociatedSite(domain) {
+  const site = await domain.getSite();
+  const branch = site[domain.context === 'site' ? 'defaultBranch' : 'demoBranch'];
+  if (branch) {
+    await Build.create({
+      user: null,
+      site: site.id,
+      branch,
+      username: 'Federalist Operators',
+    }).then(b => b.enqueue());
+  }
+}
+
+/**
+ * @param {DomainModel} domain
+ * @returns {Promise<DomainModel>}
+ */
+async function updateSiteForProvisionedDomain(domain) {
```

Comment:
does this need to requery for the site, or can we just pass in both the site and domain as arguments since the only use case right now already has them.

Reply:
There's no reason I couldn't do it the way you suggest.

Reply:
On closer inspection the only place this is called right now is in `DomainService`'s `checkProvisionStatus()`, which does _not_ already have the site. I vote for leaving the site query where it is in `updateSiteForProvisionedDomain()`.

The same goes for `updateSiteForDeprovisionedDomain()` and `checkDeprovisionStatus()`. 

What do you say, @davemcorwin?

Reply:
For the record this suggestion was about `updateSiteForProvisionedDomain`'s call of `rebuildAssociatedSite`, which is a place where the change makes sense. Taking care of it now.